src/zope/testrunner/tests/testrunner-layers-instances.txt

```diff
@@ -0,0 +1,183 @@
+Layers implemented via object instances
+=======================================
+
+Layers are generally implemented as classes using class methods, but
+regular objects can be used as well.  They need to provide __module__,
+__name__, and __bases__ attributes.
+
+>>> class TestLayer:
```

Comment:
> Should derive from object (for clarity, because we straddle Python2 / Python3).

I've seen [layers in the wild](https://github.com/zopefoundation/zope.app.testing/blob/ac2353c932bd055d0f8abcc2a74806f03d2f48ef/src/zope/app/testing/functional.py#L320) that are old-style classes on Python 2.  I wanted to be sure they continue to work.

> Should we declare that we implement IFullTestLayer? (I realize the runner doesn't check it)

Again, existing real-world usage doesn't do that and I wanted this test to be a representative example.

(Also, declaring that TestLayer is an `@implementer(IFullTestLayer)` would raise questions about doing the analoguous `@provider(...)` for class-based layers defined in the other tests.)